### PR TITLE
Add KDF bindings

### PR DESCRIPTION
**Pull request by eploko**
This is some preliminary effort to expose the KDF API in libsodium.

All KDF-related tests ported from libsodium.

I guess this is now ready for review. Please let me know if anything needs to be changed or improved for this PR to be accepted and merged.

### CODE REVIEW

File: dev-resources/vectors/README.md [hunk 1]
```diff
@@ -1 +1,14 @@
-These vectors were taken from Kalium's test suite.
+The vectors in the following folders were taken from Kalium's test suite:
```
Comment (lvh) [inline dev-resources/vectors/README.md:1]: Unfortunately that is not quite right. Comments inline.

Comment (lvh) [reply dev-resources/vectors/README.md:1]: Actually on second thought we should put READMEs in the individual source directories.

Please:
 1. add a README to `kdf` specifically mentioning where you got this
 2. delete this README

I'll add READMEs to the other directories.

**Comment by lvh** [comment]
Please consider adding a CHANGELOG entry.

**Comment by lvh** [comment]
I have confirmed that the test vectors are legit.

**Comment by lvh** [comment]
This looks great! Happy to merge after above administrative bits are taken care of.

File: dev-resources/vectors/README.md (hunk 1 above)
Comment (eploko) [reply dev-resources/vectors/README.md]: Done.

**Comment by eploko** [comment]
Merged the changes from `master` and added the separate `README.md` file in the `dev-resources/vectors/kdf` folder.

**Comment by lvh** [comment]
Looks great, thanks!

**Pull request merged**